### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by travisjeffery** [travisjeffery/jocko#1]

**Comment by sunfaces** [comment]
excellent work!

**Comment by WanliTian** [comment]
any plant for supporint push?

**Comment by bruth** [comment]
This is great. To clarify this..

> I head up the architecture at an analytics company that runs terabytes of data through Kafka everyday and I'd switch it out for Jocko.

Did you do this or do you intend to? Also do you have a list of simple tasks that early adopters or supports can help with?

**Comment by travisjeffery** [comment]
@WanliTian no. Aside from generally not wanting to break off from Kafka's API now. I agree with the [reasoning for pull in Kafka's design documentation](https://kafka.apache.org/documentation#design_pull).

@bruth I intend to. As for contributing, consumer groups would be the simplest one I can think of and I've already added the protocols needed for that feature's API. But I need to add documentation and a contributing doc to help out with that but I'm focused on replication right now.

**Comment by alexkkk** [comment]
great

**Comment by dc0d** [comment]
This is great & would be a boost for Go too! Salute!

**Comment by mlegenhausen** [comment]
Is it possible to use jocko for testing purpose right now? If yes could you provide a readme for how to start jocko?

**Comment by transitorykris** [comment]
@mlegenhausen I was trying to get it working tonight. Had to make some tweaks to work for me. Added a quick start guide for in this PR https://github.com/travisjeffery/jocko/pull/10

**Comment by travisjeffery** [comment]
Cleaned up the examples a bit: https://github.com/travisjeffery/jocko/tree/master/examples. Simple produce and fetch may work for you now given that sarama example works. You would have to create a topic like I have there — since clients don't support creating topics yet. A script for that would be a good contribution. @mlegenhausen I'd be interested to know what your outcome is, let me know in a new issue.

**Comment by curlup** [comment]
Any plans on changing High-Availability options? 

I'm asking because I don't like Kafka's take on ISR and stuff, I'm more of a fan of Cassandra Consistency level approach with Quorum as a sweet spot between HA and Consistency guarantees.

**Comment by nvartolomei** [comment]
@curlup probably you want to check uber/cherami-server.

**Comment by 17twenty** [comment]
Any plans for getting Samza to play nicely with this? Samza still wants zookeeper which means it causes issues trying to integrate the two.

**Comment by anaray** [comment]
Someone started on the ConsumerGroup impl? If yes, there are any design plans available?

**Comment by inancgumus** [comment]
I see not enough tests, one of the goals are in WIP I guess? Should help.

**Comment by LeeWong** [comment]
Any update?

**Comment by travisjeffery** [comment]
Project's [still going](https://twitter.com/travisjeffery/status/899064713325293576). 

Wrote a doc on [How you can contribute today.](https://github.com/travisjeffery/jocko/issues/48)

**Comment by alexellis** [comment]
Good work. I'm running the [OpenFaaS](https://github.com/openfaas/faas) project and working on a  Kafka integration. Jocko could make development and testing much easier. I'm just trying to figure out whether we need consumer groups for MVP.

**Comment by brendanodwyer** [comment]
@travisjeffery Is there any plans to continue work on this now that you are working at Confluent? Or is progress being left solely up to the open source community? Just curious as I stumbled across this and it sounded like an interesting project to contribute to.

**Comment by travisjeffery** [comment]
@brendanodwyer I plan to continue at some point. I'm currently writing a book (on building distributed services actually) and that takes up my free time. In any case I'll need some help and I'll help out any who are willing to put in the work.
